ipatests/test_integration/test_commands.py

```diff
@@ -133,3 +134,61 @@ def test_change_sysaccount_password_issue7561(self):
 
         tasks.ldappasswd_sysaccount_change(sysuser, original_passwd,
                                            new_passwd, master)
+
+
+class TestIPACAInstallCommand(IntegrationTest):
+    """
+    Test ipa-ca-install command and its options.
+    """
+    num_replicas = 1
+
+    @pytest.mark.xfail(reason='Ticket N 6985')
+    def test_install_ca_replica(self):
+        """
+        ipa-ca-install command installs CA on replica even if cert file
+        is not specified with --external-cert-file option. If executed
+        command with non-existing file, invalid file etc,
+        """
+        # related to https://pagure.io/freeipa/issue/6985
+        master = self.master
+
+        replica = self.replicas[0]
+        tasks.install_master(self.master)
+        tasks.install_replica(master, replica)
+
+        contents = (
+            '-----BEGIN CERTIFICATE-----\n'
+            'sdnmsdkfbsdifbsdbasdsdSDDDasdmnd\n'
+            '-----END CERTIFICATE-----')
+        cert1 = mkdtemp(suffix='xyz.crt', dir=paths.TMP)
+        replica.put_file_contents(cert1, contents)
+        cert_files = [cert1]
+
+        # install ca with invalid cert
+        result = tasks.install_ca(replica, cert_files=cert_files,
+                                  raiseonerr=False)
+        assert result.returncode != 0
+        if result.returncode == 0:
+            tasks.uninstall_replica(master, replica)
+            tasks.install_replica(master, replica)
+
+        # install ca when cert file is not specified with
+        # --external-cert-file option
+        cmd1 = ["ipa-ca-install", "-U", "-p",
+                replica.config.dirman_password, "-P", 'admin',
+                "-w", replica.config.admin_password,
+                '--external-cert-file=']
+        result1 = replica.run_command(cmd1, raiseonerr=False)
+        assert result1.returncode != 0
+
+        if result1.returncode == 0:
+            tasks.uninstall_replica(master, replica)
+            tasks.install_replica(master, replica)
```

Comment:
same comment as above, need to call tasks.install_replica(master, replica, setup_ca=False)